### DISCUSSION TITLE
Missing Metrics in Prometheus on openshift 3.11

### DISCUSSION THREAD

**Discussion by vijayibm** [prometheus/prometheus#11878] [Q&A]
### What did you do?

Hello Team,

We are running RedHat Openshift  3.11 in our both (Staging and production) environment.
Suddenly Our Grafana tool stop showing visual of node's data. Exp CPU utilization, Memory Utilization and swap etc.
While Investigation we found queries (node:node_cpu_utilisation:avg1m) is not showing data. But similar queries is working in Staging environment. Following are output of queries executed in both environment.

### What did you expect to see?

Result should be come as per working server.

### Working

[user1@ocp-st-serverA ~]$ oc exec prometheus-k8s-0 -c prometheus -- curl 127.0.0.1:9090/api/v1/query?query=node:node_cpu_utilisation:avg1m | jq  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                 Dload  Upload   Total   Spent    Left  Speed100   926  100   926    0     0   398k      0 --:--:-- --:--:-- --:--:--  904k
{  "status": "success",  "data": {    "resultType": "vector",    "result": [      {        "metric": {          "__name__": "node:node_cpu_utilisation:avg1m",          "node": "compute1"        },        "value": [          1671016052.655,          "0.047625000001668694"        ]      },      {        "metric": {          "__name__": "node:node_cpu_utilisation:avg1m",          "node": "compute2"        },        "value": [          1671016052.655,          "0.04054166666658898"        ]      },      {        "metric": {          "__name__": "node:node_cpu_utilisation:avg1m",          "node": "compute3"        },        "value": [          1671016052.655,          "0.06375000000116415"        ]      },      {        "metric": {          "__name__": "node:node_cpu_utilisation:avg1m",          "node": "master1"        },        "value": [          1671016052.655,          "0.07262499999972838"        ]      },      {        "metric": {          "__name__": "node:node_cpu_utilisation:avg1m",          "node": "master2"        },        "value": [          1671016052.655,          "0.1156250000019402"        ]      },      {        "metric": {          "__name__": "node:node_cpu_utilisation:avg1m",          "node": "master3"        },        "value": [          1671016052.655,          "0.04591666666383376"        ]      }    ]  }
}
### Not Working

[user1@ocp-pr-serverA ~]$ oc exec prometheus-k8s-0 -c prometheus -- curl 127.0.0.1:9090/api/v1/query?query=node:node_cpu_utilisation:avg1m | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    63  100    63    0     0  33210      0 --:--:-- --:--:-- --:--:-- 63000
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": []
  }
}

### What did you see instead? Under which circumstances?

Nothing. There is no data reflecting in Grafana dashboard.

### Not Working

[user1@ocp-pr-serverA ~]$ oc exec prometheus-k8s-0 -c prometheus -- curl 127.0.0.1:9090/api/v1/query?query=node:node_cpu_utilisation:avg1m | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    63  100    63    0     0  33210      0 --:--:-- --:--:-- --:--:-- 63000
{
  "status": "success",
  "data": {
    "resultType": "vector",
    "result": []
  }
}

### System information

3.10.0-1160.76.1.el7.x86_74

### Prometheus version

```text
sh-4.2$ prometheus --version
prometheus, version 2.3.2 (branch: , revision: )
  build user:       mockbuild@x86-vm-27.build.eng.bos.redhat.com
  build date:       20200818-06:11:00
  go version:       go1.10.8
```

### Prometheus configuration file

```yaml
# my global config
global:
  scrape_interval:     15s # Set the scrape interval to every 15 seconds. Default is every 1 minute.
  evaluation_interval: 15s # Evaluate rules every 15 seconds. The default is every 1 minute.
  # scrape_timeout is set to the global default (10s).
 
# Alertmanager configuration
alerting:
  alertmanagers:
  - static_configs:
    - targets:
      # - alertmanager:9093
 
# Load rules once and periodically evaluate them according to the global 'evaluation_interval'.
rule_files:
  # - "first_rules.yml"
  # - "second_rules.yml"
 
# A scrape configuration containing exactly one endpoint to scrape:
# Here it's Prometheus itself.
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: 'prometheus'
 
    # metrics_path defaults to '/metrics'
    # scheme defaults to 'http'.
 
    static_configs:
    - targets: ['localhost:9090']
```

### Alertmanager version

```text
alertmanager, version 0.15.2 (branch: , revision: )
  build user:       mockbuild@x86-017.build.eng.bos.redhat.com
  build date:       20200818-06:12:23
  go version:       go1.10.8
```

### Alertmanager configuration file

```yaml
global:
  resolve_timeout: 5m
route:
  group_wait: 30s
  group_interval: 5m
  repeat_interval: 12h
  receiver: default
  routes:
  - match:
      alertname: DeadMansSwitch
    repeat_interval: 5m
    receiver: deadmansswitch
receivers:
- name: default
- name: deadmansswitch
```

### Logs

```text
Prometheus logs

sending alert" err="Post https://10.129.1.149:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:12:44.26486015Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.141:9094/api/v1/alerts count=6 msg="Error sending alert" err="Post https://10.129.1.141:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:12:44.268987743Z caller=notifier.go:473 component=notifier alertmanager=https://10.128.0.185:9094/api/v1/alerts count=6 msg="Error sending alert" err="Post https://10.128.0.185:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:12:47.021863Z caller=notifier.go:473 component=notifier alertmanager=https://10.128.0.185:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.128.0.185:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:12:47.022705065Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.141:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.129.1.141:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:12:47.025764658Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.149:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.129.1.149:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:00.18025592Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.141:9094/api/v1/alerts count=3 msg="Error sending alert" err="Post https://10.129.1.141:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:00.18370525Z caller=notifier.go:473 component=notifier alertmanager=https://10.128.0.185:9094/api/v1/alerts count=3 msg="Error sending alert" err="Post https://10.128.0.185:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:00.185702756Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.149:9094/api/v1/alerts count=3 msg="Error sending alert" err="Post https://10.129.1.149:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:00.191535917Z caller=notifier.go:473 component=notifier alertmanager=https://10.128.0.185:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.128.0.185:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:00.191800795Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.149:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.129.1.149:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:00.199393807Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.141:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.129.1.141:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:02.049289256Z caller=notifier.go:473 component=notifier alertmanager=https://10.128.0.185:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.128.0.185:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:02.056335959Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.149:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.129.1.149:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:02.060071272Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.141:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.129.1.141:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:04.4936868Z caller=notifier.go:473 component=notifier alertmanager=https://10.128.0.185:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.128.0.185:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"
level=error ts=2022-12-14T19:13:04.498797787Z caller=notifier.go:473 component=notifier alertmanager=https://10.129.1.149:9094/api/v1/alerts count=1 msg="Error sending alert" err="Post https://10.129.1.149:9094/api/v1/alerts: x509: certificate has expired or is not yet valid"




alertmanager logs

level=info ts=2022-11-06T01:54:44.27325772Z caller=main.go:174 msg="Starting Alertmanager" version="(version=0.15.2, branch=, revision=)"
level=info ts=2022-11-06T01:54:44.27344799Z caller=main.go:175 build_context="(go=go1.10.8, user=mockbuild@x86-017.build.eng.bos.redhat.com, date=20200818-06:12:23)"
level=info ts=2022-11-06T01:54:44.363647623Z caller=main.go:322 msg="Loading configuration file" file=/etc/alertmanager/config/alertmanager.yaml
level=info ts=2022-11-06T01:54:44.364324078Z caller=cluster.go:570 component=cluster msg="Waiting for gossip to settle..." interval=2s
level=info ts=2022-11-06T01:54:44.367058922Z caller=main.go:398 msg=Listening address=127.0.0.1:9093
level=info ts=2022-11-06T01:54:46.36481692Z caller=cluster.go:595 component=cluster msg="gossip not settled" polls=0 before=0 now=3 elapsed=2.000399921s
level=info ts=2022-11-06T01:54:54.366421977Z caller=cluster.go:587 component=cluster msg="gossip settled; proceeding" elapsed=10.002006971s
level=error ts=2022-11-14T17:32:53.288893719Z caller=api.go:813 msg="API error" err="bad_data: unexpected EOF"
```

**Comment by roidelapluie** [comment]
Hi Vijayibm,

Thanks for your request. To troubleshoot this issue, I would recommend checking if the Prometheus targets are up and running correctly in the production environment. You can check the status of the targets by running the command "oc exec prometheus-k8s-0 -c prometheus -- curl 127.0.0.1:9090/api/v1/targets" or executing the "up" query.

Also, please check the recording rules configuration, make sure that the rules that are working in the staging environment are also configured in the production environment.

Please note that this is more of a support request than an issue, so I am moving this to a discussion.
